framework/infrastructure/gcp/cloud_run.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2025 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import abc
+import logging
+
+from googleapiclient import discovery
+
+from framework.infrastructure import gcp
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_TEST_PORT = 8080
+DEFAULT_TIMEOUT = 600
+
+
+class CloudRunApiManager(
+    gcp.api.GcpStandardCloudApiResource, metaclass=abc.ABCMeta
```

Comment:
This is a new-style API, it should be extending `GcpProjectApiResource`. I think this may fix the logic of waiting on the delete operation too.

Reply:
Ah, no, I'm wrong here. `GcpStandardCloudApiResource` is the right class here.